analyzers/tests/SonarAnalyzer.UnitTest/TestCases/InfiniteRecursion.Roslyn.cs

```diff
@@ -0,0 +1,357 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Tests.Diagnostics
+{
+    class InfiniteRecursion
+    {
+        int Pow(int num, int exponent)   // Noncompliant; no condition under which pow isn't re-called
+//          ^^^
+        {
+            num = num * Pow(num, exponent - 1);
+            return num;  // this is never reached
+        }
+
+        void Test1() // Noncompliant {{Add a way to break out of this method's recursion.}}
+        {
+            var i = 10;
+            if (i == 10)
+            {
+                Test1();
+            }
+            else
+            {
+                switch (i)
+                {
+                    case 1:
+                        Test1();
+                        break;
+                    default:
+                        Test1();
+                        break;
+                }
+            }
+        }
+
+        void Test2()
+        {
+            var i = 10;
+            switch (i)
+            {
+                case 1:
+                    goto default;
+                default:
+                    goto case 1; // Compliant FN
+            }
+        }
+
+        void Test3()
+        {
+            var i = 10;
+            switch (i)
+            {
+                case 1:
+                    goto default;
+                case 2:
+                    break;
+                default:
+                    goto case 1; // Compliant FN
```

Comment:
In general, this doesn't comply with the RSPEC so it's not compliant. It's just a FN
```suggestion
                    goto case 1; // FN
```

Reply:
These were fixed in Sonar flavor, please fix them in Roslyn one as well.

Reply:
What do you mean by that? I've applied the suggestion in Roslyn. In Sonar they were not fixed, but supported by the previous implementation since 2017.

Reply:
Roslyn test, lines 268 and 271 - I want to change `Compliant FN` comment only to `FN`, because the case doesn't comply with the RSPEC